app/src/main/java/org/mozilla/fenix/search/telemetry/Utils.kt

```diff
@@ -33,7 +48,7 @@ internal fun getTrackKey(
         }
 
         // For Bing if it didn't have a valid cookie and for all the other search engines
-        if (hasValidCode(code, provider)) {
+        if (hasValidCode(originalCode, provider)) {
```

Comment:
Another option is to to call `uri.getQueryParameter(provider.codeParam)` directly here and then we won't need `originalCode`.

Reply:
I think this could be the easiest way.